v2/client/client.go

```diff
@@ -36,17 +36,18 @@ type Client interface {
 	// This call is blocking.
 	// Valid fn signatures are:
 	// * func()
-	// * func() error
 	// * func(context.Context)
-	// * func(context.Context) protocol.Result
 	// * func(event.Event)
-	// * func(event.Event) protocol.Result
 	// * func(context.Context, event.Event)
-	// * func(context.Context, event.Event) protocol.Result
 	// * func(event.Event) *event.Event
-	// * func(event.Event) (*event.Event, protocol.Result)
 	// * func(context.Context, event.Event) *event.Event
-	// * func(context.Context, event.Event) (*event.Event, protocol.Result)
+	// These signatures are available with an additional error returned:
+	// * func() error
+	// * ...
+	// * func(context.Context, event.Event) (*event.Event, error)
+	// The error returned may impact the messages processing made by the protocol
+	// used (example: message acknowledgement). Please refer to each protocol's
+	// package documentation of the function "Finish(err error) error".
```

Comment:
- replaced all protocol.Result by simple error
- removed all functions with additional error returned and added a dedicated notice about ability to returned an additional error for al cases
- made a reference to each protocol's Finish doc to detail acknowledgement behaviours.

wdyt?

Reply:
LGTM

Reply:
For clarity I'd prefer the old style of having the individual function signatures (with and without error). The "..." could be misinterpreted and become outdated if we add another signate without an error type. So my recommendation would be to switch back to the previous "explicit" style, use `error` as you have and then finish the comment with your note with "The error returned may...". OK?